week-3/Homework/A-array-find/exercise.js

```diff
@@ -4,10 +4,23 @@
 */
 
 // write your code here
+function findLongNameThatStartsWithA(name) {
+  if (name.length > 7 && name.startsWith("A")) return name;
+}
 
-var names = ["Rakesh", "Antonio", "Alexandra", "Andronicus", "Annam", "Mikey", "Anastasia", "Karim", "Ahmed"];
+var names = [
+  "Rakesh",
+  "Antonio",
+  "Alexandra",
+  "Andronicus",
+  "Annam",
+  "Mikey",
+  "Anastasia",
+  "Karim",
+  "Ahmed",
+];
 
-var longNameThatStartsWithA = findLongNameThatStartsWithA(names);
+var longNameThatStartsWithA = names.find(findLongNameThatStartsWithA);
 
```

Comment:
This gives the right result but it isn't really a good way of doing it. The find function expects a callback that returns a boolean (i.e. true or false) and when it gets true it then returns the corresponding value of the array, not the value that was returned by your callback function works.

The reason your function still works in practice is because Javascript has a concept called 'truthyness' as well as 'falsyness'. It's quite complicated, but in this case what it means is that when it sees any string where it might expect a boolean, it treats it the same as the boolean value of `true`. When nothing is returned at all it sees it as `false`. That's why your code still gives the right result. However you should always seek to return a `true` or `false` when a boolean is expected.

Reply:
Thanks for your feedback. I understand how it should work. My function will return the result of statement like `return name.length > 7 && name.startsWith("A");`
Thank you so much.